src/mem/Stats.cc

```diff
@@ -29,5 +24,12 @@ Mem::GlobalStats(PoolStats &stats)
         stats.overhead += sizeof(Allocator *);
     }
 
+    // Reset PoolStats::meter, label, and obj_size data members after getStats()
+    // calls in the above loop set them. TODO: Refactor to remove these members.
```

Comment:
Refactoring mentioned in the above TODO will produce a lot of noise, so I did not include it in this bug-fixing PR.

I did remove PoolStats::pool member because keeping a stale value would essentially preserve a part of the bug this PR is fixing while resetting that raw pointer to nil is dangerous (some future code can easily start dereferencing it). That member removal did not produce a lot of noise because the removed data member was barely used.

P.S. This comment does not request any PR changes.

Reply:
Please clarify in the code comment what you mean by "these members".

FTR; this particular `stats` variable holds the details for the entire `MemPools` pooling structure. The meta-pool of all pools (`stats.overhead += sizeof(MemPools)`) and allocations in terms of bytes (`stats.obj_size = 1`) .

Reply:
Well, going by that clarification, I think nobody else will be able to enact it without a much more detailed knowledge of what you in particular think is wrong with them. For now they serve a clear purpose.

So ... remove the TODO and enact it in a future PR if/when you have time to figure out the details.